src/daft-core/src/array/ops/list.rs

```diff
@@ -384,6 +384,95 @@ impl ListArray {
         ))
     }
 
+    pub fn distinct(&self) -> DaftResult<Self> {
+        use std::collections::HashMap;
+
+        let flat_child = self.flat_child.to_arrow();
+        let flat_child = &*flat_child;
+
+        let is_equal = build_is_equal(
+            flat_child, flat_child,
+            false, // this value does not matter; invalid (= nulls) are never included
```

Comment:
If I'm not wrong, we can set this to `true` then skip the special casing for validity in your main loop